statchatui/src/Home/Home.jsx

```diff
@@ -1,19 +1,75 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import Stats from "../Stats/Stats";
 import "./Home.css";
 
 const GAME_NAMES = {
   APEX: "Apex Legends",
   R6: "Rainbow Six Siege",
 };
-const Home = ({ onStatsChange, stats }) => {
-  //FIXME initial state should store user's info from the database if it already exists
+async function fetchStats(formData) {
+  const apexApiKey = process.env.REACT_APP_API_KEY;
+  const baseUrl = "https://api.mozambiquehe.re/bridge?auth=";
+  if (
+    formData.gamertag === "" ||
+    formData.platform === "" ||
+    formData.gameName === ""
+  ) {
+    return null;
+  }
+  try {
+    let response;
+    if (formData.gameName === GAME_NAMES.APEX) {
+      response = await fetch(
+        `${baseUrl}${apexApiKey}&player=${formData.gamertag}&platform=${formData.platform}`
+      );
+    } else if (formData.gameName === GAME_NAMES.R6) {
+      response = await fetch(
+        `https://api.henrikdev.xyz/r6/v1/profile/${formData.platform.toLowerCase()}/${
+          formData.gamertag
+        }`
+      );
+    }
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error submitting form:", error);
+  }
+}
+const Home = ({ onStatsChange, stats, isSignedIn }) => {
   const [formData, setFormData] = useState({
     gamertag: "",
     platform: "",
     gameName: "",
   });
 
+  const getUserInfo = async () => {
+    try {
+      if (isSignedIn) {
+        const profileResponse = await fetch(
+          `http://localhost:3002/users/profile`,
+          {
+            method: "GET",
+            credentials: "include",
+          }
+        );
+        const profileData = await profileResponse.json();
+        setFormData({
+          gamertag: profileData.gamertag,
+          platform: profileData.platform,
+          gameName: profileData.gameName,
+        });
+        const stats = await fetchStats(profileData);
+        onStatsChange(stats);
+      }
+    } catch (error) {
+      console.error("Error fetching user info:", error);
+    }
+  };
+
+  useEffect(() => {
+    getUserInfo();
+  }, [isSignedIn]); // calls getuserinfo when the page initiallly loads and only when it initially loads.
```

Comment:
This comment isn't quite right. Since this is in a useEffect, we call getUserInfo here whenever `isSignedIn` changes. Actually we should do a check inside the useEffect:

```
  if (isSignedIn) {
    getUserInfo();
  }
```